sre-recipes/recipes/encoding_recipe/encoding_recipe.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -*- coding: utf-8 -*-
+"""
+This module contains the implementation of recipe 1
+"""
+
+import logging
+import subprocess
+from recipe import Recipe
+
+
+class EncodingRecipe(Recipe):
+    """
+    This class implements recipe 1, which purposefully
+    spits errors from the Email Service.
+    """
+
+    def deploy_state(state):
+        """
+        Sets an environment variable ENCODE_EMAIL to given state
+        and updates the state accordingly
+        """
+        state_str = str(state).lower()
+        set_env_command = f"kubectl set env deployment/emailservice ENCODE_EMAIL={state_str}"
+        get_pod_command = """kubectl get pod -l app=emailservice -o \
+            jsonpath=\"{.items[0].metadata.name}\""""
+        logging.info('Setting env variable: %s', set_env_command)
+        logging.info('Getting pod: %s', get_pod_command)
+
+        Recipe._run_command(set_env_command)
+        service, error = Recipe._run_command(get_pod_command)
```

Comment:
It's not a super high priority, but it would be nice to see error handling. There are a lot of places where things could go wrong here, but the code always assumes success. What if a user runs `sandboxctl` after deleting their project? Will the errors they see make sense?

Reply:
Good point - I added error handling into the CLI refactor PR

Reply:
Hah I pranked myself. Added error handling for the deploy state method and for the hint method